### DISCUSSION TITLE
Profiling data incomplete

### DISCUSSION THREAD

**Discussion by aplund** [jax-ml/jax#31655] [Q&A]
I'm trying to figure out why the profiler gives me somehow incomplete information.  I'm running the following:

```
import jax
options = jax.profiler.ProfileOptions()
options.python_tracer_level = 1
options.host_tracer_level = 2
options.device_tracer_level = 1

with jax.profiler.trace(
    "/tmp/jax-trace",
    create_perfetto_link=True,
    create_perfetto_trace=True,
    profiler_options=options,
):
    arr = jax.numpy.arange(100)
    tr = jax.numpy.trace(jax.numpy.reshape(arr, (10,10)))
    tr.block_until_ready()
```

The perfetto interface shows only three threads, the main python one and two tf_xla-cpu-codegen threads.  But nothing about the actual dispatched computation.  There's also a warning about loss of data, which I can believe.  But it doesn't give any more detail.

Is the size of the tracing buffer somehow limited?

<img width="174" height="101" alt="20250909_15h34m48s_grim" src="https://github.com/user-attachments/assets/abe29682-453a-4b9a-a8fa-7c0156168568" />

<img width="762" height="896" alt="20250909_15h37m50s_grim" src="https://github.com/user-attachments/assets/f6b303ac-e5c1-4656-8bfe-8ccea5a15a0d" />

[perfetto_trace.json.gz](https://github.com/user-attachments/files/22227122/perfetto_trace.json.gz)
